packages/aws-cdk-lib/aws-route53/lib/record-set.ts

```diff
@@ -245,6 +270,16 @@ export class RecordSet extends Resource implements IRecordSet {
   constructor(scope: Construct, id: string, props: RecordSetProps) {
     super(scope, id);
 
+    if (props.weight && (props.weight < 0 || props.weight > 255)) {
+      throw new Error(`weight must be between 0 and 255 inclusive, got: ${props.weight}`);
+    }
+    if ((props.setIdentifier && (props.setIdentifier.length < 1 || props.setIdentifier.length > 128) || props.setIdentifier === '')) {
+      throw new Error(`setIdentifier must be between 1 and 128 characters long, got: ${props.setIdentifier.length}`);
+    }
+    if (props.weight && !props.setIdentifier) {
```

Comment:
Great! I've added a check for when they are defined simultaneously.

Reply:
@lpizzinidev 
It seems that generating `setIdentifier` based on `weight` alone might not be sufficient to handle identical weights. However, creating an ID that includes the `target` seems like it would guarantee uniqueness. 
May I proceed with the implementation in that way?
```ts
new route53.ARecord(this, `WeightedRecord${index}`, {
        zone: hostedZone,
        recordName: 'www',
        weight: 20,
        ttl: Duration.seconds(10),
        target: route53.RecordTarget.fromIpAddresses('1.2.3.4'),
      }); 
// -> 'A-www-1.2.3.4'
```

Reply:
Thanks for clarifying.
Not sure about using the `target` since users can specify lists of values and it could be difficult to format.
Also, `name` has a max length of 1024 so it could be a problem to use given that `setIdentifier` has a 128 characters limit.
What about keeping it simple: 
```ts
const idPrefix = `WEIGHT_${weight}_ID_`;
const identifier = `${idPrefix}${Names.uniqueResourceName(this, { maxLength: 64 - idPrefix.length })}`
```
If users need a specific `setIdentifier` they can always use the newly exposed property.